superdesk/publish/http_push.py

```diff
@@ -0,0 +1,156 @@
+            raise PublishHTTPPushError.httpPushError(ex, queue_item.get('destination', {}))
+
+    def _get(self, url, item_id=None):
+        '''Retrieve an item from the given resource url
+        Raises ItemNotFound if the item identified by item_id (if given) did not exist
+        Raises Exception on other errors
+
+        @param url: the resource url
+        @param item_id: the item identifier
+        @return: HTTPResponse object
+        '''
+        if item_id:
+            url = url.format(item_id)
+        request_client = self._request_client(url)
+        request_client.request('GET', url)
+        response = request_client.getresponse()
+        if response.status == 404:
+            raise ItemNotFound()
+        if response.status != 200:
+            raise Exception('Error retrieving url %s: %s' % (url, response.reason))
+        return response
+
+    def _post(self, url, formatted_item):
+        '''Send a post request to content API
+        Raises IdConflict if the item identifier already existed
+        Raises Exception on other errors
+
+        @param url: the resource url
+        @param formatted_item: string containing the item in the required format
+        @return: HTTPResponse object
+        '''
+        request_client = self._request_client(url)
+        request_client.request('POST', url, formatted_item, self.headers)
+        response = request_client.getresponse()
+        if response.status == 409:
+            raise IdConflict()
+        if response.status != 201:
+            raise Exception('Error pushing item %s' % response.reason)
+        return response
+
+    def _headers(self, *custom_headers):
+        '''Return headers for an HTTP request
+
+        @param custom_headers: tuples of header-value pairs
+        '''
+        updated_headers = copy(self.headers)
+        updated_headers.update(custom_headers)
+        return updated_headers
+
+    def _patch(self, item_url, _id, formatted_item, etag):
+        '''Send a patch request to content API to update a given item
+        Raises ItemNotFound if the item identified by _id did not exist
+        Raises Exception on other errors
+
+        @param item_url: the format of the item url
+        @param _id: the item identifier
+        @param formatted_item: string containing the item in the required format
+        @param etag: the etag of the existing item
+        @return: HTTPResponse object
+        '''
+        url = item_url.format(_id)
+        headers = self._headers(self.headers, ('If-Match', etag))
+        request_client = self._request_client(url)
+        request_client.request('PATCH', url, formatted_item, headers)
+        response = request_client.getresponse()
+        if response.status == 404:
+            raise ItemNotFound()
+        if response.status < 200 or response.status >= 300:
+            raise Exception('Error pushing item %s' % response.reason)
+        return response
+
+    def _request_client(self, url):
```

Comment:
imo we can use [requests](http://www.python-requests.org/en/latest/) here it would make it simpler